CRM/Contribute/Form/ContributionPage/TabHeader.php

```diff
@@ -142,7 +142,14 @@ public static function process(&$form) {
     switch ($className) {
       case 'Contribute':
         $attributes = $form->getVar('_attributes');
-        $class = strtolower(basename(CRM_Utils_Array::value('action', $attributes)));
+        $rawUrl = CRM_Utils_Array::value('action', $attributes);
```

Comment:
@christianwach we seem to be repeating the same code 4 places - maybe add a function on CRM_Utils_Request class or CRM_Utils_String (I think the former has the url() function) & call that?

Reply:
@eileenmcnaughton Fair point. I had considered this but wasn't sure where to locate such a method - thanks for the suggestion.